core/streamdata.go

```diff
@@ -21,9 +23,33 @@ const (
 	HashLength = 32
 )
 
+type SegmentMetadata struct {
```

Comment:
Can we leave a comment here explaining why we are tracking ManifestID instead of StreamID?  (Already explained in the PR notes, but want to make sure it stays with the code)

Reply:
How should we explain it? The PR notes work for reviewers who understand the previous state of the codebase and need to relate to how it's changed, but it seems a little strange to refer to "previous state" for comments in an updated codebase.

Presumably the PR note is this one:

> The ManifestID is used here rather than the StreamID since it is a little bit shorter, and the additional (often incorrect) information carried by the appended rendition string is not needed.

Reply:
I think we want to explain the semantics of what it means to have a ManifestID and multiple VideoProfile in SegmentMetadata.  Theoretically, a video segment can ONLY have 1 VideoProfile.  It looks like we are only use `SegmentMetadata` to signal transcoding preference.  I don't want someone to misuse this for something else (for example, representing metadata for a specific segment)

Reply:
@j0sh and I talked about renaming `SegmentMetadata` to `SegTranscodingMetadata`.  Can we do that?

Reply:
Done